Parser/Tree_Nodes/while_node.c

```diff
@@ -23,15 +23,57 @@
  * @param next - Lookahead pointer
  * @return new While node with a i_expr conditional and b_stmt_list
  */
-while_node *create_while_node(i_expr *conditional, b_stmt_list *body) {
+while_node *create_while_node(GArray *token_stream, unsigned long index, unsigned long *next) {
     while_node *retVal = NULL;
 
     // Make new While node
     retVal = calloc(1, sizeof(while_node));
-    retVal->body = body;
-    retVal->conditional = conditional;
-    //Return the value
-    return retVal;
+    (*next)++;
+    // Load next token
+    Token *check = &g_array_index(token_stream, Token, *next);
```

Comment:
You should be using index, not next. Next is really only used after you create something with an unknown length. For example, when creating an assigmnment, we create the expression, then use next to make sure the following token is a semicolon